### ISSUE TITLE
increasing max_history vs. epochs ?

### ISSUE THREAD

**Issue by znat** [RasaHQ/rasa_nlu#1925]
I noticed that as the dialog flow deepens/widens (more branches, more turns), increasing `max_history` or `epochs` both increase accuracy.

I am trying to understand the implication of both.

My understanding, for now, is that increasing `max_history` enriches the model by letting it learn from a longer sequence of dialogue turns (provided examples are long enough), whereas increasing `epochs` improves the memorization of the training and overfits. 

Is that correct?
